### PR TITLE
start placement & cert controller only when mco CR available

### PR DESCRIPTION
**Pull request by marcolan018**
https://github.com/open-cluster-management/backlog/issues/13263

Signed-off-by: Marco <llan@redhat.com>

### CODE REVIEW

File: controllers/multiclusterobservability/multiclusterobservability_controller.go [hunk 1]
```diff
@@ -106,6 +110,14 @@ func (r *MultiClusterObservabilityReconciler) Reconcile(ctx context.Context, req
 	// start to update mco status
 	StartStatusUpdate(r.Client, instance)
 
+	if os.Getenv("UNIT_TEST") != "true" {
+		// start placement controller
+		placemengctrl.StartPlacementController(r.Manager)
+
+		// setup ocm addon manager
+		certctrl.Start(r.Client)
+	}
```
Comment (clyang82) [inline controllers/multiclusterobservability/multiclusterobservability_controller.go:119]: They will be started many times in reconciler w/o control

Comment (marcolan018) [reply controllers/multiclusterobservability/multiclusterobservability_controller.go:119]: fixed

File: controllers/placementrule/placementrule_controller.go [hunk 2]
```diff
@@ -585,3 +587,36 @@ func (r *PlacementRuleReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	// create and return a new controller
 	return ctrBuilder.Complete(r)
 }
+
+func StartPlacementController(mgr manager.Manager) error {
+	if isplacementControllerRunnning {
+		return nil
+	}
+	isplacementControllerRunnning = true
```
Comment (clyang82) [inline controllers/placementrule/placementrule_controller.go:595]: Do you think we need to cover this case? create CR --> start placement controller --> delete CR --> stop placement controller ---> create CR again --> start placement controller.

Comment (marcolan018) [reply controllers/placementrule/placementrule_controller.go:595]: @clyang82 
imo, it's not necessary. placement controller will do nothing if no mco cr.

**Pull request merged**